record/record_test.go

```diff
@@ -332,22 +332,52 @@ func writeRecords(records [][]byte) ([]byte, error) {
 	return buf.Bytes(), nil
 }
 
-func TestBasicRecover(t *testing.T) {
-	records := [][]byte{
-		[]byte(strings.Repeat("a", blockSize-headerSize)),
-		[]byte(strings.Repeat("b", blockSize-headerSize)),
-		[]byte(strings.Repeat("c", blockSize-headerSize)),
+// generateTestRecords returns up to 256 records in a slice of slices of the specified lengths.
+// The first record will be repeating 0x00 bytes, the next 0x01 bytes, and so forth.
+func generateTestRecords(recordLengths []int) [][]byte {
```

Comment:
It's probably easier if you changed
(recordLengths []int)
to
(recordLengths ...int)